hw/top_earlgrey/rdc/rdc_waivers.spi_device.tcl

```diff
@@ -101,3 +101,20 @@ set_rule_status -rule {E_RDC_METASTABILITY} -status {Waived} \
   -expression {(SourceReset=~"*u_spi_device.rst_csb_buf") && \
     (ObservableFlopReset=~"*u_reg.u_control_mode*")} \
   -comment {CONTROL.mode is configured at the init time.}
+
+
+# CPHA, CONTROL.mode waivers for main rst
```

Comment:
sorry i'm really confused by this one.
If we issue the IP reset or its POK, doesn't every flop in the block get reset..? 
I didn't quite follow what we mean by if SPI line is stalled, something might not get reset.

Reply:
<img width="1126" alt="image" src="https://user-images.githubusercontent.com/1192814/199837880-8ef84070-f265-451b-9f9f-1b114ba8c889.png">

The RDC tool does not show the waveform but only schematic. So the waveform above is my assumption (maybe wrong)

When CPHA is set to 1, the spi_in, spi_out clocks are inverted, when the IP receives the POK or SwRst, the CPHA bit goes bace to 0. By going back to 0, the clock now has an edge (either posedge or negedge). That moves the logic forward if CSb is 0.

I suspect that's why the tool reports errors on the signal. If we can assume CSb is 1 at the time POK or SwRst is issued, we can ignore this error.

Reply:
hmmm... that could be a hard assumption to make though.
Because if we decide to reset the block, it is possible we are doing so because the host crashed, and maybe the host `CSb` is stuck at 0 or something. 

We probably could reset the host through a request, but it would be nice if it were more self contained.

One last thing to clarify though, the edge that gets created due to the polarities resetting, doesn't that still only impact logic that is still under reset..? Or are you saying we have logic that is only reset by CSb and not the normal `rst_ni`?

Reply:
huh.. why I kept thinking that CSb reset is not AND-ed with sys_rst_ni (even COMBO_RST warning reports that path)? .. you are right. We can safely ignore this for any SwRst, POR_N cases.

Hum, then, I don't know why the tool reports errors on these paths.. :(

Reply:
the skeptical side of me wants to say the tool is stupid... sigh..
should we double check we didn't accidentally have a CSb only reset flop?

Reply:
I just checked the design. Fortunately no logic only reset by CSb (or TPM CSb)

Reply:
Ah.. I defined wrong reset. I should specifi `rst_spi_n` and `tpm_rst_n` as resets but I described `rst_csb_buf` and `rst_tpm_csb_buf` as resets. They are the signal prior to OR-ed by sys reset. 

I don't know this is the right one or not. Maybe... (should ask FAE)